### PR TITLE
schedstatistics: Convert to ztimer

### PR DESCRIPTION
**Pull request by bergzand**
### Contribution description

Another ztimer conversion, this time `sys/schedstatistics`

### Testing procedure

there is a test in `tests/ps_schedstatistics`

### Issues/PRs references

Another one for #17111

### CODE REVIEW

**Review by kaspar030 — APPROVED** [review]
LGTM.

**Comment by bergzand** [comment]
Also includes conversion of `tests/periph_spi` now because it uses internal schedstatistics members.

**Comment by fjmolinas** [comment]
> Also includes conversion of `tests/periph_spi` now because it uses internal schedstatistics members.

Can you show some log on the spi part?

**Comment by bergzand** [comment]
> @bergzand are there no updates required on Kconfig?

grml, might have forgotten to update the `app.config.test`

**Comment by bergzand** [comment]
> grml, might have forgotten to update the `app.config.test`

Added, thanks for the heads-up.

**Comment by fjmolinas** [comment]
@bergzand 
```diff
diff --git a/tests/periph_spi_dma/app.config.test b/tests/periph_spi_dma/app.config.test
index 29981de4c5..e00e33a81b 100644
--- a/tests/periph_spi_dma/app.config.test
+++ b/tests/periph_spi_dma/app.config.test
@@ -6,4 +6,6 @@ CONFIG_MODULE_PERIPH_SPI=y
 CONFIG_MODULE_SHELL=y
 CONFIG_MODULE_SHELL_COMMANDS=y
 CONFIG_MODULE_SCHEDSTATISTICS=y
-CONFIG_MODULE_XTIMER=y
+CONFIG_MODULE_ZTIMER=y
+CONFIG_MODULE_ZTIMER_SEC=y
+CONFIG_ZTIMER_USEC=y
```

**Comment by fjmolinas** [comment]
And blacklisting:

```
--- FAILED build outputs:
--- build output of app tests/periph_spi for board samd10-xmini:gnu (raw, runtime=0.5s):
-- running on worker mobi6.inet.haw-hamburg.de-17 thread 1, build number 1692.
make: Entering directory '/tmp/dwq.0.045125125006728495/61d9b9ef168c13ef99eaa7037570e517/tests/periph_spi'
Building application "tests_periph_spi" for "samd10-xmini" with MCU "samd21".
```

**Comment by bergzand** [comment]
> @bergzand

done, thanks

> And blacklisting:

and done.

**Comment by bergzand** [comment]
and squashed.

File: sys/schedstatistics/Kconfig [hunk 1]
```diff
@@ -7,6 +7,6 @@
 
 config MODULE_SCHEDSTATISTICS
     bool "Scheduler statistics support"
-    depends on MODULE_XTIMER
+    depends on MODULE_ZTIMER_USEC
```
Comment (fjmolinas) [inline sys/schedstatistics/Kconfig:10]:
```suggestion
    select ZTIMER_USEC
```

File: tests/periph_spi_dma/app.config.test [hunk 2]
```diff
@@ -6,4 +6,6 @@ CONFIG_MODULE_PERIPH_SPI=y
 CONFIG_MODULE_SHELL=y
 CONFIG_MODULE_SHELL_COMMANDS=y
 CONFIG_MODULE_SCHEDSTATISTICS=y
-CONFIG_MODULE_XTIMER=y
+CONFIG_MODULE_ZTIMER=y
+CONFIG_MODULE_ZTIMER_SEC=y
+CONFIG_MODULE_ZTIMER_USEC=y
```
Comment (fjmolinas) [inline tests/periph_spi_dma/app.config.test:11]:
```suggestion
CONFIG_ZTIMER_USEC=y
```

Comment (bergzand) [reply tests/periph_spi_dma/app.config.test]: Changed!

**Review by fjmolinas — APPROVED** [review]
ACK, tested on `nucleo-l152re` works as expected:

```
2021-12-07 20:21:39,151 # 	pid | name                 | state    Q | pri | stack  ( used) ( free) | base addr  | current     | runtime  | switches  | runtime_usec 
2021-12-07 20:21:39,181 # 	  - | isr_stack            | -        - |   - |    512 (  232) (  280) | 0x20000000 | 0x200001c8
2021-12-07 20:21:39,229 # 	  1 | main                 | running  Q |   7 |   1536 (  808) (  728) | 0x2000182c | 0x20001b0c  |  2.170% |     19530  |    2145751 
2021-12-07 20:21:39,271 # 	  2 | thread               | bl mutex _ |   6 |   1024 (  412) (  612) | 0x2000042c | 0x20000744  | 10.463% |     60596  |   10345876 
2021-12-07 20:21:39,319 # 	  3 | thread               | bl mutex _ |   6 |   1024 (  412) (  612) | 0x2000082c | 0x20000b44  | 15.066% |     60626  |   14898045 
2021-12-07 20:21:39,367 # 	  4 | thread               | bl mutex _ |   6 |   1024 (  412) (  612) | 0x20000c2c | 0x20000f44  | 19.675% |     60656  |   19454675 
2021-12-07 20:21:39,415 # 	  5 | thread               | bl mutex _ |   6 |   1024 (  412) (  612) | 0x2000102c | 0x20001344  | 24.287% |     60686  |   24015775 
2021-12-07 20:21:39,457 # 	  6 | thread               | bl mutex _ |   6 |   1024 (  412) (  612) | 0x2000142c | 0x20001744  |  5.878% |     60716  |    5812808 
2021-12-07 20:21:39,481 # 	    | SUM                  |            |     |   7168 ( 3100) ( 4068)
```

**Pull request merged**